net/FlatBuffers/ByteBuffer.cs

```diff
@@ -50,12 +54,16 @@ namespace FlatBuffers
 {
     public abstract class ByteBufferAllocator : IDisposable
     {
-#if UNSAFE_BYTEBUFFER
-        public unsafe byte* Buffer
```

Comment:
I think it is worth noting that with this change you would no longer be able to use a custom allocator without also enabling Span<T> support.  This would really only be an issue for older versions of .net that do not support Span<T>.  Custom allocators was only recently added and I don't expect anyone will run into this, but I thought it should be 
noted.
Sorry, for not mentioning it earlier but I just realized it while reviewing the code.

Reply:
Is there somewhere we should make a note of it?

Reply:
Note that the documentation at the top of the file already indicates this, so I think we're OK?

https://github.com/google/flatbuffers/blob/4f32cbf268a8ad469974b9453924a3582dbf2f30/net/FlatBuffers/ByteBuffer.cs#L19-L31

Reply:
Looks good to me!